src/actions/downloadMetadataFromUri.ts

```diff
@@ -0,0 +1,20 @@
+import { DomainMetadata, IPFSGatewayUri } from "../types";
+import { makeApiCall } from "../api/actions/helpers";
+
+export const downloadMetadataFromUri = async (
+  metadataUri: string,
+  ipfsGatewayUri: IPFSGatewayUri
+): Promise<DomainMetadata> => {
+  let metadata: DomainMetadata;
```

Comment:
Consider putting this code in a helper so it doesn't have to be repeated here and in `getDomainMetadata`, the only difference is that this takes a uri and `getDomainMetadata` takes a `domainId` and then calls to the registrar to find it.